pkg/build/strategies/sti/sti.go

```diff
@@ -250,6 +248,15 @@ func (b *STI) Determine(request *api.Request) (err error) {
 		return
 	}
 
+	// if image does not exist in local registry try searching for it
+	if !previousImageExists && request.ForcePull {
```

Comment:
why are you checking forcepull here?  forcepull is to force pulling the sti base image, right?  we shouldn't overload it's meaning.

also according to the trello card i think the behavior should be:
1) default to clean build (do not pull previous image)
2) otherwise you always need to pull the previous image (even if it exists, it could be stale)